ethcontract-common/src/artifact.rs

```diff
@@ -0,0 +1,97 @@
+//!
+
+use crate::errors::ArtifactError;
+use crate::Contract;
+
+/// An entity that contains compiled contracts.
+pub trait Artifact {
```

Comment:
I like this abstraction. If only there were HKT you could use an `Iter` associated type :stuck_out_tongue:.